### ISSUE TITLE
Unable to add a condition for an alias (SELECT BB AS XXX WHERE XXX > 10)

### ISSUE THREAD

**Issue by anthosz** [influxdata/influxdb#9913]
Hello,

InfluxDB: 1.5.2

**It seems that we cannot add an arithmetic condition (on integer) when we use alias (the alias seems to be forced to string).**

When do a simpe subquery it is working as expected and the result is the following one:
Query 

- (works):

`SELECT used_memory FROM "redis" where used_memory > 1`

- Results:

> time                used_memory 
> ----                -----
> 1527501630000000000 295193800
> 1527501630000000000 484549808

`SELECT maxmemory FROM "redis" where maxmemory > 1`

- Results:

> time                maxmemory 
> ----                -----
> 1527501750000000000 419430400
> 1527501750000000000 1258291200

Now same query but with an alias:

Query (not works):

`SELECT used_memory AS aa FROM "redis" where aa > 1`

- Results:

(no results, empty response)

- Query (not works, try to convert/force to integer):

`SELECT maxmemory AS aa FROM "redis" where aa::integer > 1`

- Results:

(no results, empty response)

- Query (all fields seems to be ok):

`SHOW FIELD KEYS`

- Results:

> maxmemory                      integer
> used_memory                    integer

So basically, I cannot add a filter on a query with an arithmetic operation.

The goal is to execute this query:
`SELECT (last("used_memory") / last("maxmemory")) * 100 AS ratio FROM "redis" WHERE ratio > XX (replace XX by integer)`

**Comment by anthosz** [comment]
Hello,

Any news on that?

Could you be able to reproduce the issue?

Let me know if you need more information

Best regards

**Comment by teeson** [comment]
I had the same problem, I try to do these but fail:
`SELECT difference(val) AS diff FROM test WHERE diff >= 0`
but always return nothing

Is there any alternatives or plan to solve these question?

**Comment by anthosz** [comment]
Same issue with infludb 1.7.2

@dgnorton will be fixed in 2.x?

**Comment by anthosz** [comment]
Up
